tc/core/tensor-inl.h

```diff
@@ -0,0 +1,322 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <algorithm>
+#include <sstream>
+
+#include <glog/logging.h>
+
+namespace tc {
+
+template <typename DLTensorType>
+inline void setSizes(DLTensorType& t, const std::vector<int64_t>& sizes) {
```

Comment:
which code style are you using here? Normally we use Google code style which uses pointers for mutable arguments. And they go in the end of the list of function inputs. This is also the most common style at Facebook AFAIK

Reply:
<rant>
I would strongly argue against passing pointers due to  "historical reasons", especially in a new project. A pointer can be null, but in this particular we never want it to be null. So we either introduce a check and possibly throw at runtime, or we use a reference which makes explicit the fact that we expect a fully constructed object. 

Yes, a user can go out of her/his way to construct a reference out of a `nullptr` but then it's their fault.
</rant>

Reply:
I always hate that discussion, both sides have reasonable arguments.
So I'll just be removing that useless code. 
In general and as much as possible we favor return types.